comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go

```diff
@@ -232,7 +232,7 @@ func (c *collector) parsePods(
 
 		var nsLabels, nsAnnotations map[string]string
 
-		if c.dcaClient.SupportsNamespaceMetadataCollection() {
+		if c.isDCAEnabled() && c.dcaClient.SupportsNamespaceMetadataCollection() {
```

Comment:
Should the `else` also check `c.isDCAEnabled()` ?

Reply:
No, it is already checked again [here](https://github.com/DataDog/datadog-agent/blob/6b3d202f342cb84b0ab329b4287a792dd77647cd/comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go#L254). 

Specifically, the check is [here](https://github.com/DataDog/datadog-agent/blob/6b3d202f342cb84b0ab329b4287a792dd77647cd/comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go#L321).